docs/src/main/sphinx/functions/json.md

```diff
@@ -1138,6 +1138,227 @@ FROM customers
+
+`NESTED PATH` extracts data from nested levels of a `json_input` value. Each
+`NESTED PATH` clause can contain `column_definition` values.
+
+The `json_table` function returns a result set that you can use like any other
+table in your queries. You can join the result set with other tables or
+combine multiple arrays from your JSON data.
+
+You can also process nested JSON objects without parsing the data multiple
+times.
+
+Use `json_table` as a lateral join to process JSON data from another table.
+
+### Examples
+
+The following query uses `json_table` to extract values from an array of JSON
+objects and return them as rows in a table with three columns:
+
+```sql
+SELECT
+      *
+FROM
+      json_table(
+                '[
+                  {"id":1,"name":"Africa","wikiDataId":"Q15"},
+                  {"id":2,"name":"Americas","wikiDataId":"Q828"},
+                  {"id":3,"name":"Asia","wikiDataId":"Q48"},
+                  {"id":4,"name":"Europe","wikiDataId":"Q51"}
+                ]',
+                'strict $' COLUMNS (
+                  NESTED PATH 'strict $[*]' COLUMNS (
+                    id integer PATH 'strict $.id',
+                    name varchar PATH 'strict $.name',
+                    wiki_data_id varchar PATH 'strict $."wikiDataId"'
+                  )
+                )
+              );
+```
+
+| id | child     | wiki_data_id  |
+| -- | --------- | ------------- |
+|  1 | Africa    | Q1            |
+|  2 | Americas  | Q828          |
+|  3 | Asia      | Q48           |
+|  4 | Europe    | Q51           |
+
+The following query uses `json_table` to extract values from an array of nested
+JSON objects:
+
+```sql
+SELECT
+      *
+FROM
+      json_table(
+                '[
+                    {"continent": "Asia", "countries": [
+                        {"name": "Japan", "population": 125.7},
+                        {"name": "Thailand", "population": 71.6}
+                    ]},
+                    {"continent": "Europe", "countries": [
+                        {"name": "France", "population": 67.4},
+                        {"name": "Germany", "population": 83.2}
+                    ]}
+                ]',
+                'lax $' COLUMNS (
+                    NESTED PATH 'lax $[*]' COLUMNS (
+                        continent varchar PATH 'lax $.continent',
+                        NESTED PATH 'lax $.countries[*]' COLUMNS (
+                            country varchar PATH 'lax $.name',
+                            population double PATH 'lax $.population'
+                        )
+                    )
+                ));
+```
+
+| continent  | country   | population    |
+| ---------- | --------- | ------------- |
+| Asia       | Japan     | 125.7         |
+| Asia       | Thailand  | 71.6          |
+| Europe     | France    | 67.4          |
+| Europe     | Germany   | 83.2          |
+
+The following query uses `PLAN` to specify an `OUTER` join between a parent path
```

Comment:
Given that the next two examples use and empty array of input and the whole syntax is not really explained I think we either have to update these now .. or revisit in a follow up PR (I prefer that so we can ship this PR) .. but as it stands I think the two examples are basically a headscratcher for the reader and barely useful unless you know and understand already.